### DISCUSSION TITLE
Z2M restart after LAN adapter back online

### DISCUSSION THREAD

**Discussion by eugenjy** [tteck/Proxmox#2352] [Help Needed] [answered]
Hi! I have a server with z2m lxc, debian 12. The Zigbee LAN adapter is remotely connected. When the adapter is powered off, z2m stops. But when it is back online, I must log into the server and start z2m manually. Can you please help with a script to start z2m after the adapter IP ping is successful?

**Comment by tteck** [comment, marked as answer]
Maybe something like this?

Create a script to check and start the Zigbee2MQTT service if it's not running.
```
cat <<EOF >/opt/zigbee2mqtt/check_zigbee_service.sh
if ! systemctl is-active --quiet zigbee2mqtt.service; then
  systemctl start zigbee2mqtt.service
fi
EOF
```
Give it execute permissions:
```
chmod +x /opt/zigbee2mqtt/check_zigbee_service.sh
```
Open your crontab file for editing:
```
crontab -e
```

Add a line to schedule the script to run at your desired interval. For example, to run every 5 minutes:
```
*/5 * * * * /opt/zigbee2mqtt/check_zigbee_service.sh
```
This has not been tested, but _should_ work.

**Comment by 3vilson** [comment]
Thanks got it woking on Alpine LXC.

Create the directory:
```
mkdir /opt/zigbee2mqtt
```
Create a script to check if the Zigbee2MQTT service crashed then execute a reboot.
```
cat <<EOF >/opt/zigbee2mqtt/check_zigbee_service.sh
if rc-service zigbee2mqtt status | grep -q "started"; then
    echo "zigbee2mqtt is running."
else
    echo "zigbee2mqtt is not running. Restarting..."
    #rc-service zigbee2mqtt restart
	reboot
fi
EOF
```
Give it execute permissions:
```
chmod +x /opt/zigbee2mqtt/check_zigbee_service.sh
```

Open your crontab file for editing:
```
nano /etc/crontabs/root
```
Add a line to schedule the script to run at your desired interval. For example, to run every 5 minutes:
```
*/5		*	*	*	*	/opt/zigbee2mqtt/check_zigbee_service.sh
```

**reboot**

**Reply by aperezva** [reply]
Please

could you confirm it’s works?
